tokenizer/src/lib.rs

```diff
@@ -1,595 +1,300 @@
 use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
-use near_sdk::collections::{LookupMap, Vector};
-use near_sdk::json_types::{U128};
 use near_sdk::{
-    env, near_bindgen, AccountId, Balance, Promise, StorageUsage
+    env, near_bindgen, AccountId, Balance, PromiseResult
 };
 
-const TOKENIZER: &[u8] = b"TOKENIZER";
+/**
+ * external contract interface load.
+ */ 
+// use token::{nep21};
+// use registry::{registry};
+use token::ScaleToken;
+use registry::Registry;
+pub mod utils;
+mod validator;
+
+#[global_allocator]
+static ALLOC: near_sdk::wee_alloc::WeeAlloc<'_> = near_sdk::wee_alloc::WeeAlloc::INIT;
+
+const DEPOSIT_AND_STAKING_GAS: u64 = 100_000_000_000_000;
 const STORAGE_PRICE_PER_BYTE: Balance = 100_000_000_000_000_000_000;
+const SINGLE_CALL_GAS: u64 = 200_000_000_000_000;
 
 #[near_bindgen]
 #[derive(BorshDeserialize, BorshSerialize)]
 pub struct Tokenizer {
+    token: AccountId,
     governance: AccountId,
-    registry: Vector<Registry>,
+    registry: AccountId,
+}
+
+impl Default for Tokenizer {
+    fn default() -> Self {
+        panic!("Tokenizer should be initialized before usage")
+    }
 }
 
+#[near_bindgen]
 impl Tokenizer {
-    pub fn new(governor: AccountId) -> Self {
-        let index_key_id = append(&TOKENIZER, b'l');
+
+    #[init]
+    pub fn new(governance: AccountId, registry: AccountId) -> Self {
+        assert!(!env::state_exists(), "Registry: Already initialized");
         Self {
-            governance: governor,
-            registry: Vector::new(index_key_id)
+            token: "".to_string(),
+            governance,
+            registry
+        }
+    }
+
+    pub fn set_token(&mut self, token: AccountId) {
+        let caller = env::predecessor_account_id();
+        if caller != self.governance {
+            env::panic(b"Caller is not Governance");
         }
+        self.token = token;
     }
 
-    pub fn add_registry(&mut self, validator: AccountId, ratio: u32) {
-        self.registry.push(&Registry::new(validator, ratio))
+    pub fn deposit(&self) {
+        let owner_id = env::predecessor_account_id();
+        let deposit = env::attached_deposit();
+
+        let receipt = env::promise_create(
+            self.registry.clone(),
+            b"get_validators",
+            &[],
+            0,
+            SINGLE_CALL_GAS,
+        );
+
+        let validators_bytes = match env::promise_result(receipt) {
+            PromiseResult::Successful(x) => x,
+            PromiseResult::Failed => env::panic(b"The promise failed. See receipt failures."),
+            PromiseResult::NotReady => env::panic(b"The promise was not ready."),
+        };
+
+        let validator_info = Vec<>::From(&validators_bytes);
```

Comment:
`get_validators` it should be returned `Vec<(AccountId, u32)>` could be convert? it seems to be `Vec<u8>`